### PR TITLE
Save the exit block, if any, in CFGTraversal. NFC

### PR DESCRIPTION
**Pull request by kripken**
This will be used (and tested) in dead store elimination.

### CODE REVIEW

File: src/cfg/cfg-traversal.h [hunk 1]
```diff
@@ -45,7 +45,13 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
     std::vector<BasicBlock*> out, in;
   };
 
-  BasicBlock* entry; // the entry block
+  // The entry block at the function's start. This always exists.
+  BasicBlock* entry;
+
+  // The exit block at the end where control flow leaves the function. This may
```
Comment (tlively) [inline src/cfg/cfg-traversal.h:51]: Maybe "reaches the end" rather than "leaves" to make it clearer that `return` doesn't count.

**Review by tlively — APPROVED** [review]

File: src/cfg/cfg-traversal.h [hunk 2]
```diff
@@ -45,7 +45,13 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
...
+  // not exist if nothing flows out directly (for example, if the function ends
+  // in a return or an unreachable).
+  BasicBlock* exit;
```
Comment (aheejin) [inline src/cfg/cfg-traversal.h:54]: - If a function ends in a return, is that not considered "flowing out directly"?
- What if there are multiple exit points?

File: src/cfg/cfg-traversal.h (hunk 1 above)
Comment (kripken) [reply src/cfg/cfg-traversal.h:51]: Ah, that is better, thanks.

File: src/cfg/cfg-traversal.h (hunk 2 above)
Comment (kripken) [reply src/cfg/cfg-traversal.h]: I improved the comment now, PTAL. I agree it was not clear enough before. It should address your questions here I hope.

**Review by aheejin — APPROVED** [review]
Thanks for the detailed comment!

**Pull request merged**